### ISSUE TITLE
Error when compiling: SDL2 library, required by Sdl2Application, was not found. 

### ISSUE THREAD

**Issue by Luc45** [tiltedphoques/TiltedEvolution#562]
When running both `xmake project -k vsxmake` or `xmake -y` following the [Build Guide](https://wiki.tiltedphoques.com/tilted-online/technical-documentation/build-guide), I get the following error message:

![image](https://user-images.githubusercontent.com/9341686/213889062-1ea092a1-f3f1-47ba-9f1e-cf75e5eabe31.png)

The [xmake.lua file](https://github.com/tiltedphoques/TiltedEvolution/blob/master/xmake.lua#L48-L50) in the root of the project explicitly ask for SDL2 when installing `magnum`:

```lua
add_requireconfs("magnum", { configs = { sdl2 = true }})
add_requireconfs("magnum-integration.magnum",  { configs = { sdl2 = true }})
```

I have tried [downloading the SDL2 library](https://github.com/libsdl-org/SDL/releases/tag/release-2.26.2) directly and placing `SDL2.dll` in the root of the project, but got the same result.

In my case, I just want to compile it to use the `dev` branch to see if I can minimize crashing by leveraging some of the fixes in this branch. I'd appreciate any guidance in compiling.

Alternatively, if someone could provide me with a compiled version of both Client and Server with the latest commit in `dev` branch, that would be appreciated!

Thanks.

**Comment by Luc45** [comment]
I also tried forking this repo and uncommenting the "Build "and "Upload Artifact" steps in the "Build Windows" GitHub Workflow, but it failed: https://github.com/Luc45/TiltedEvolution/actions/runs/3977305511/jobs/6818347083

**Comment by stenliis** [comment]
Error in https://github.com/Luc45/TiltedEvolution/actions/runs/3977305511/jobs/6818347083 may be resolved by PR #563.

Issue with SDL library sound to me similar to the one fixed in PR #551.
Did you try build project on dev branch that contains this PR?

**Comment by Dragonisser** [comment]
Ive just recently downloaded the dev branch and compiled it without any issues. What version of xmake are xou using.

**Comment by Luc45** [comment]
>  What version of xmake are you using

FIrst I used latest 2.7.5, then I downgraded to 2.6.5 because this is what the GitHub Workflows were using.

> Did you try build project on dev branch that contains this PR

I did. I do not want to bother, but I get a different error. It's weird because it doesn't seem to be coming from the project, but rather my installation of xmake.

 With `-v` (Verbose)  and `-D` (Backtrace) enabled:

![image](https://user-images.githubusercontent.com/9341686/213919367-d84a0321-f833-4440-b5f3-eb3fb98cd50d.png)

> Error in https://github.com/Luc45/TiltedEvolution/actions/runs/3977305511/jobs/6818347083 may be resolved by PR https://github.com/tiltedphoques/TiltedEvolution/pull/563.

This worked. The compiled artifact was generated by the Workflow: https://github.com/Luc45/TiltedEvolution/actions/runs/3979974640

Overall, I think we can close this issue by #551. Thanks a lot for your help, folks.

PS: I don't want to rush at all, but just to know, is there a timeline in mind for merging `dev` work into `master` and releasing a new stable update?
